src/opts.c

```diff
@@ -367,12 +367,10 @@ natsOptions_SetSecure(natsOptions *opts, bool secure)
 natsStatus
 natsOptions_TLSHandshakeFirst(natsOptions *opts)
 {
-    natsStatus s = NATS_OK;
-
     LOCK_AND_CHECK_OPTIONS(opts, 0);
 
     opts->tlsHandshakeFirst = true;
-    opts->secure            = true;
+    natsStatus s = natsOptions_SetSecure(opts, true);
```

Comment:
I know that we started to define variables anywhere in the code now, and I guess I should be ok with that as long as compilers (with our given settings) don't complain, but if you don't mind, leave the `natsStatus s;` defined above and here have:
```
s = natsOptions_SetSecure(opts, true);
if (s == NATS_OK)
    opts-tlsHandshakeFirst = true;
```